src/metaswitch/clearwater/etcd_shared/common_etcd_synchronizer.py

```diff
@@ -224,7 +224,21 @@ def read_from_etcd(self, wait=True):
             _log.info("Key {} doesn't exist in etcd yet".format(self.key()))
             # Sleep briefly to avoid hammering a non-existent key.
             sleep(self.PAUSE_BEFORE_RETRY_ON_MISSING_KEY)
-            return (self.default_value(), None)
+            # Use any value on disk first, but the default value if not found
+            try:
+                f = open(self._plugin.file(), 'r')
+                value = f.read()  # pragma: no cover
+            except:
+                value = self.default_value()
+            # Attempt to create new key in etcd.
+            try:
+                self._client.write(self.key(), value, prevExist=False)
```

Comment:
What does the PrevExist parameter do? Can you comment why this is a safe thing to do (e.g. it will only write if the key doesn't already exist, so there's no race condition where this overwrites a value a user has entered).